### PR TITLE
proxsuite 0.2.2 (new formula)

### PR DESCRIPTION
**Pull request by jcarpent**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
It fails on ARM (Apple Silicon) because it includes a x86-only header:

```
14/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/mmintrin.h::214:: 2error: :"This header is only meant to be used on x86 and x64 architecture" 
```

And on Intel, the test fails:

```
  ==> Testing proxsuite
  ==> /usr/local/opt/python@3.10/libexec/bin/python -c import proxsuite
  qp = proxsuite.proxqp.dense.QP(10,0,0)
  assert qp.model.H.shape[0] == 10 and qp.model.H.shape[1] == 10
  Traceback (most recent call last):
  Error: test failed
    File "<string>", line 3, in <module>
  ModuleNotFoundError: No module named 'numpy'
```

**Comment by jcarpent** [comment]
@fxcoudert @carlocab I've fixed the issue with @fabinsh to properly integrate ProxSuite within HomeBrew.
This PR is ready for review.

**Comment by jcarpent** [comment]
Thanks a lot @fxcoudert @carlocab for your time and help.

File: Formula/proxsuite.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    ENV.prepend_path "PYTHONPATH", Formula["eigenpy"].opt_prefix/Language::Python.site_packages(python3)
+    ENV.prepend_path "Eigen3_DIR", Formula["eigen"].opt_share/"eigen3/cmake"
+
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DPYTHON_EXECUTABLE=#{python3}",
+                    "-DBUILD_UNIT_TESTS=OFF",
+                    "-DBUILD_PYTHON_INTERFACE=ON",
+                    "-DINSTALL_DOCUMENTATION=ON",
+                    "-DCMAKE_OSX_DEPLOYMENT_TARGET=10.15",
```
Comment (SMillerDev) [inline Formula/proxsuite.rb:36]: Why?

File: Formula/proxsuite.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    "-DSimde_INCLUDE_DIR=#{Formula["simde"].opt_prefix/"include"}",
```
Comment (SMillerDev) [inline Formula/proxsuite.rb:37]: Why doesn't CMake detect this correctly?

File: Formula/proxsuite.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    Formula["python@3.10"].opt_libexec/"bin/python"
+  end
+
+  def install
+    if build.head?
+      system "git", "submodule", "update", "--init"
+      system "git", "pull", "--unshallow", "--tags"
+    end
+
+    ENV.prepend_path "PYTHONPATH", Formula["eigenpy"].opt_prefix/Language::Python.site_packages(python3)
```
Comment (SMillerDev) [inline Formula/proxsuite.rb:28]: This isn't in your dependencies

File: Formula/proxsuite.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  depends_on "pkg-config" => :build
+  depends_on "eigen"
+  depends_on "numpy"
+  depends_on "python@3.10"
+  depends_on "scipy"
+  depends_on "simde"
+
+  def python3
+    Formula["python@3.10"].opt_libexec/"bin/python"
+  end
```
Comment (SMillerDev) [inline Formula/proxsuite.rb:20]: Since it's only used once this shouldn't be a function

Comment (jcarpent) [reply Formula/proxsuite.rb:20]: OK. I've handled it.

File: Formula/proxsuite.rb (hunk 2 above)
Comment (jcarpent) [reply Formula/proxsuite.rb:37]: I don't know. But this seems to not work here. Is it a problem?

**Comment by jcarpent** [comment]
Thanks @SMillerDev for the feedback. I've handled most of them, except the one to properly find Simde.

File: Formula/proxsuite.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/proxsuite.rb:31]: Can you file an issue upstream to make them aware of this?

Comment (jcarpent) [reply Formula/proxsuite.rb:31]: Yes, will do.

**Comment by jcarpent** [comment]
@SMillerDev Thanks a lot for the review process.
Do you have in mind a timeline for releasing this package in the core of Homebrew?

**Comment by SMillerDev** [comment]
When it's done and reviewed

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
LGTM

File: Formula/proxsuite.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/proxsuite.rb:31]: Still waiting for that issue

Comment (jcarpent) [reply Formula/proxsuite.rb:31]: I've just opened an issue on this here: https://github.com/Simple-Robotics/proxsuite/issues/65

File: Formula/proxsuite.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  def install
+    if build.head?
+      system "git", "submodule", "update", "--init"
+      system "git", "pull", "--unshallow", "--tags"
+    end
+
+    ENV.prepend_path "PYTHONPATH", Formula["eigenpy"].opt_prefix/Language::Python.site_packages
+
+    system "cmake", "-S", ".", "-B", "build",
```
Comment (SMillerDev) [inline Formula/proxsuite.rb:26]:
```suggestion
    # simde include dir can be removed after https://github.com/Simple-Robotics/proxsuite/issues/65
    system "cmake", "-S", ".", "-B", "build",
```

Comment (jcarpent) [reply Formula/proxsuite.rb:27]: Thanks a lot @SMillerDev for the comment.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**